### PR TITLE
Kailey Smith [DS-Unit-2-Sprint-3-Advanced-Regression]

### PR DESCRIPTION
**Pull request by krsmith**

### CODE REVIEW

**Comment by jojordan3** [comment]
## Sprint Challenge Code Review
### Unit 2 - Sprint 3: Advanced Regression
---
#### Part 0 — Run this starter code
---
#### Part 1 — Load, Validate, and Prepare Data
<details><summary><em>Code review</em></summary>
<p>

_What went well:_

- Glad to see yo got those column titles and did some exploration regarding NaNs

- Incorporating unknowns in your categorical encoding was a good move.

_What could be better:_

- Not really a critique, but I'd be interested to know how you decided what to map each value to in your encoding. Generally, when label encoding you would want some sort of order. For logistic regression and most other types of models (besides linear regression), one hot encoding would be a better choice. ski-kit and pandas both have built in functions to do these things.

sci-kit (sklearn)
[Label Encoding (keeps the same column, just puts numbers into the values like you did)](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html
)
[One Hot Encoding (makes a new binary column for each category for the feature)](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html)

pandas
[super useful, pandas has its own `get dummies` function which is its version of one hot encoding](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.get_dummies.html)

- More comments saying what you're doing and why would be good here

</p>
</details>

---
#### Part 2 — Fit and Present a Logistic Regression Model
<details><summary><em>Code review</em></summary>
<p>

_What went well:_

- Glad the first thing you did was print out a metric to evaluate the function.

- Confusion matrix is a great way to analyze a binary logistic regression

- I like your visualization about education vs income in the bar chart.

_What could be better:_

- When it comes to binary classification where the data is unbalanced (many more of one outcome than of another--i.e. there are way more people who made less than 50k) accuracy isn't actually a great metric for evaluation. There are many, but an F1 score is a good one. We'll probably learn about that next week.

- In addition, the best use of the confusion matrix is to determine the number of false positives vs false negatives. Again, we'll prob talk about this next week so don't worry too much for now.

</p>
</details>

---
#### Part 3 — Analysis, Interpretation, and Questions: Model Analysis and Interpretations
<details><summary><em>Code review</em></summary>
<p>

_What went well:_

- Generally solid answers and good analysis

_What could be better:_

- Would be good to include actual values for the answers to 1 and 2

- Only other thing would be about Q3 related to my comments for part 2--stay tuned... haha

</p>
</details>

---
#### Part 4 — Analysis, Interpretation, and Questions: Matched Situations
<details><summary><em>Code review</em></summary>
<p>

_What went well:_

- You matched everything correctly, with solid explanations

_What could be better:_

- Could have gone further with explanations, I guess... but it's good.

</p>
</details>
